httpd/handler.go

```diff
@@ -30,6 +30,13 @@ const (
 	DefaultChunkSize = 10000
 )
 
+const (
+	// type of authentication
+	UserAuth    = iota
+	SecretAuth  = iota
+	UnknownAuth = iota
```

Comment:
You don't need iota for each one. You can do this:

```go
const (
	// type of authentication
	UserAuth = iota
	SecretAuth
	UnknownAuth
)
```

See [play example](http://play.golang.org/p/NUp36C-7Ar)